kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/BankParametersAccessServiceImpl.java

```diff
@@ -0,0 +1,759 @@
+    }
+
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#isAllowedBai(int)
+     */
+    public boolean isAllowedBai(int bai) {
+        return getAllowedBaiTypes().contains(bai);
+    }
+
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#getChartBankObjectCodeForDescriptionSearch(String)
+     */
+    public ChartBankObjectCode getChartBankObjectCodeForDescriptionSearch(String description) {
+
+        for (String descKey : getDescriptionSearchChartBankObjectCodeMap().keySet()) {
+            if (description.contains(descKey)) {
+                return descriptionSearchChartBankObjectCodeMap.get(descKey);
+            }
+        }
+        //not found, returning null;
+        return null;
+    }
+
+    /*
+     * Internal method that populates a Map with description as key and ChartBankObjectCode as values from the ACCOUNT_AND_OBJECT_CODE_BY_DESCRIPTION parameter
+     */
+    protected Map<String, ChartBankObjectCode> getDescriptionSearchChartBankObjectCodeMap() {
+        if (descriptionSearchChartBankObjectCodeMap == null) {
+            descriptionSearchChartBankObjectCodeMap = new HashMap<String, ChartBankObjectCode>();
+            String acountObjByDescrParm = parameterService.getParameterValueAsString(DocumentCreationStep.class, KFSConstants.BankTransactionsParameters.ACCOUNT_AND_OBJECT_CODE_BY_DESCRIPTION);
+            String[] acountObjByDescrList = acountObjByDescrParm.split(KFSConstants.BankTransactionsParameters.BANK_PARAMETER_DELIM);
+            if (acountObjByDescrList == null || acountObjByDescrList.length == 0) {
+                LOG.error("Parameter [ACCOUNT_AND_OBJECT_CODE_BY_DESCRIPTION] has an empty value.  The process cannot be continued without this value.");
+                throw new RuntimeException("Parameter [ACCOUNT_AND_OBJECT_CODE_BY_DESCRIPTION] has an empty value.  The process cannot be continued without this value.");
+            }
+
+            int index = 0;
+            for (String chartBankObjectCodeMapStr : acountObjByDescrList) {
+                String[] tokens = chartBankObjectCodeMapStr.split("=");
+                if (tokens == null || tokens.length != 2) {
+                    LOG.error("Parameter [ACCOUNT_AND_OBJECT_CODE_BY_DESCRIPTION] value [" + index + "] has a misconfigured value.  Expecting a string delimeted by equals-sign, got none.");
+                    throw new RuntimeException("Parameter [ACCOUNT_AND_OBJECT_CODE_BY_DESCRIPTION] value [" + index + "] has a misconfigured value.  Expecting a string delimeted by equals-sign, got none.");
+                }
+                String descrKey = tokens[0];
+                String value = tokens[1];
+                ChartBankObjectCode cbo = new ChartBankObjectCode();
+                String[] valueTokens = value.split(KFSConstants.BankTransactionsParameters.BANK_PARAMETER_DELIM);
+                if (valueTokens == null || valueTokens.length != 3) {
```

Comment:
Another split, where valueTokens cannot be null (and exception would drop us out of this method).